bootstrap_salt/ec2.py

```diff
@@ -52,38 +53,41 @@ def get_instance_by_id(self, inst_id):
         resv = self.conn_ec2.get_all_reservations([inst_id])
         return [i for r in resv for i in r.instances][0] if resv else None
 
-    def get_master_instance(self, stack_name_or_id, master_tag_name='SaltMaster'):
-        stack_instances = self.cfn.get_stack_instances(stack_name_or_id)
-        stack_instance_ids = [x.instance_id for x in stack_instances]
+    def get_master_instance(self, master_tag_name='SaltMaster'):
         filters = {'tag-key': master_tag_name,
-                   'instance-state-name': 'running',
-                   'instance-id': stack_instance_ids}
```

Comment:
Removing this filter I think means that we look at all instances in the account, not just the ones in the right stack (i.e. if we had dev and prod stacks in the account we could pick the master from the wrong stack)